index.rb

```diff
@@ -0,0 +1,114 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    0.upto(length - 1) do |i|
+      yield self[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    0.upto(length - 1) do |i|
+      yield(self[i], i)
+    end
+    self
+  end
+
+  def my_select
+    array = []
+    return to_enum unless block_given?
+
+    0.upto(length - 1) do |i|
+      array << self[i] if yield self[i]
+    end
+    array
+  end
+
+  def my_all?(patt = nil)
+    if block_given?
+      my_each { |i| return false unless yield(i) }
+    elsif patt.nil?
+      my_each { |i| return false unless i }
+    else
+      my_each { |i| return false unless check_patt(i, patt) }
+    end
+    true
+  end
```

Comment:
* [ ] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1,2,3,4,5].my_all?(3) != [1,2,3,4,5].all?(3)
```